dbt/adapters/snowflake/connections.py

```diff
@@ -63,6 +72,58 @@
 }
 
 
+@cache
```

Comment:
nit:  can we move these to their own module? Just want to keep connections.py small

Reply:
I actually want to move them into `dbt-adapters` long term. I thought about creating a separate module, but thought it would be unnecessary if we were going to move it later. I can add that here though if you think it's worth it.

Reply:
I think it's worth it as it should make it cleaner to move later

Reply:
I moved over only the two methods that are generic. I left the one that's Snowflake-specific (converting the key to DER format).

Along the same lines of making `connections.py` smaller, I actually think the `Credentials` class should be in its own module (probably named `auth`, even though that's what I moved these methods to). That's a separate PR, but it shouldn't take long to do if we want to do that.